nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -568,6 +573,26 @@ ObstacleLayer::raytraceFreespace(
     double a = wx - ox;
     double b = wy - oy;
 
+    double distance = std::sqrt(a * a + b * b);
```

Comment:
This entire block from L576 to L595 can be deleted. If you change the arguments for `raytraceLine` to take in a maximum and a minimum range (converted into cell distances),  then [after this line](https://github.com/ros-planning/navigation2/blob/d3fc28af28df764a153c85434f8b30270e963622/nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d.hpp#L414) you have a distance you can check if its lower than to exit without calling `bresenham2D`. 

Do the same with `updateRaytraceBounds`